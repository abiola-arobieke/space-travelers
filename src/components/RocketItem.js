const RocketItem = () => {
  console.log('hi');
  return (
    <div>
      <h1>This is rocket Item</h1>
    </div>
  );
};

export default RocketItem;
