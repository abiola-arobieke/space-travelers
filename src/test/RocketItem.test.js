import React from 'react';
import renderer from 'react-test-renderer';
import { useDispatch } from 'react-redux';
import RocketItem from '../components/RocketItem';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));

describe('RocketItem', () => {
  it('renders correctly', () => {
    const dispatchMock = jest.fn();
    useDispatch.mockReturnValue(dispatchMock);

    const rocket = {
      id: 1,
      name: 'Falcon 9',
      description: 'Sample description',
      reserved: false,
      flickr_images: ['https://example.com/image.jpg'],
    };

    const reservationMock = jest.fn();
    const cancelReservationMock = jest.fn();

    const component = renderer.create(
      <RocketItem
        rocket={rocket}
        reservation={reservationMock}
        cancelReservation={cancelReservationMock}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
