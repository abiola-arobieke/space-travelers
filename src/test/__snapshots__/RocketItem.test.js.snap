// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketItem renders correctly 1`] = `
<div
  className="ds-flex card-container"
>
  <div
    className="mid-screen-4"
  >
    <div
      className="card-image"
    >
      <img
        alt=""
        className="rocket-img"
        src="https://example.com/image.jpg"
      />
    </div>
  </div>
  <div
    className="mid-screen-8 card-details"
  >
    <h2
      className="card-name"
    >
      Falcon 9
    </h2>
    <p
      className="card-description"
    >
      Sample description
    </p>
    <button
      className="btn-add"
      onClick={[Function]}
      type="button"
    >
      Add Reservation
    </button>
  </div>
</div>
`;
